tempesta_fw/t/functional/leaks/test_leak.py

```diff
@@ -0,0 +1,110 @@
+"""
+Testing for memory leaks
+"""
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2017 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+import os.path
+import subprocess
+
+import psutil
+from time import sleep
+
+import unittest
+from testers import stress
+from helpers import tf_cfg, control, tempesta
+
+
+class LeakTest(stress.StressTest):
+        """ Leaks testing """
+        config = 'cache 0;\n'
+        backend_connections = 10
+        memory_leak_thresold = 16*1024 # in kib
+
+        def read_kmemleaks(self):
+                """Get amount of kmemleak records"""
+                kmemleakfile = "/sys/kernel/debug/kmemleak"
+                if not os.path.isfile(kmemleakfile):
+                        return -1
+                echocmd = "cat %s | grep \"unreferenced object\" | wc -l" % kmemleakfile
+                proc = subprocess.Popen(["sh", "-c", echocmd], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
The test is going to work only if TempestFW and test framework are running on the same host. In all other cases the test will provide useless data with possible false positives. Please see `remote.py` to see, how to run the command on target host (`control.py/Tempesta`, `control.py/Nginx` and many other classes use that approach)

Reply:
fixed